File: src/pratice/pointtopoint/좋다1253.java
package pratice.pointtopoint;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class 좋다1253 {
    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static int N;
    public static int[] A;

    public static void main(String[] args) throws IOException {
        input();
        Arrays.sort(A);
        pro();
    }

    private static void pro() {
        long cnt = 0 ;
        for (int i = 0; i < N; i++) {
            if(func(i)){
                cnt++;
            }
        }
        System.out.println(cnt);
    }

    private static boolean func(int idx) {
        int L = 0;
        int R = N - 1;
        int target = A[idx];

        while(L < R){
            if(L == idx) L++;
            else if(R == idx)R--;
            else{
                if(A[L] + A[R] > target) R--;
                else if(A[L] + A[R] < target) L++;
                else return true;
            }
        }
        return false;
    }

    private static void input() throws IOException {
        N = Integer.parseInt(br.readLine());
        A = new int[N];
        String[] split = br.readLine().split(" ");

        for (int i = 0; i < N; i++) {
            A[i] = Integer.parseInt(split[i]);
        }
    }
}

File: src/pratice/pointtopoint/좋다1253.kt
package pratice.pointtopoint

fun main() {
    val N = readln().toInt()
    val arr = readln().split(" ").map { it.toInt() }.toMutableList()
    var cnt = 0
    arr.sort()

    for (x in 0 until N) {
        if (func(x, arr)) {
            cnt++
        }
    }
    println(cnt)
}

private fun func(x: Int, arr: MutableList<Int>): Boolean {
    var L = 0
    var R = arr.size - 1
    val target = arr[x]

    while (L < R) {
        if (L == x) L++
        else if (R == x) R--
        else {
            if (arr[L] + arr[R] > target) R--
            else if (arr[L] + arr[R] < target) L++
            else return true
        }
    }
    return false
}
